File: kattis/beekeeper.c
#include <stdio.h>
#include <string.h>

char *vowels = "aeiouy";
int isVowel(char c) {
	for(int i = 0; i < 6; i++) {
		if (c == vowels[i]) return 1;
	}
	return 0;
}

int main() {
	int N, i, j, curr, best;
	char c, last;
	char fav[81];
	char job[81];
	for(;;) {
		scanf("%d", &N);
		if (N == 0) break;

		*fav = '\0';
		for(i = 0; i < N; i++) {
			scanf("%s", job);
			curr = 0;
			last = '\0';
			j = 0;
			for(c = job[0]; c != '\0'; c = job[++j]) {
				if (c == last && isVowel(c)) 
					curr++;
				last = c;
			}
			if (curr >= best) {
				strcpy(fav, job);
			}
		}

		printf("%s\n", fav);
	}
}

File: kattis/Beekeeper.java
import java.util.*;

public class Beekeeper {

	static boolean isVowel(char c) {
		return (c == 'a' || c == 'e' || c == 'i' || 
				c == 'o' || c == 'u' || c == 'y');
	}
			

	public static void main(String[] args) {
		Scanner in = new Scanner(System.in);

		for(;;) {
			int N = in.nextInt();
			if (N == 0) break;

			int best = -1;
			String favorite = "";
			for(int i = 0; i < N; i++) {
				String job = in.next();
				char last = job.charAt(0);
				int vowelCount = 0;
				for(int j = 1; j < job.length(); j++) {
					char c = job.charAt(j);
					if (c == last && isVowel(c)) {
						vowelCount += 1;
					}
					last = c;
				}
				if (vowelCount > best) {
					best = vowelCount;
					favorite = job;
				}
			}
			System.out.println(favorite);
		}
	}
}

